### PR TITLE
Default  EgressListener services should also be merged

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
For https://github.com/istio/istio/issues/49900。 because in rds builder we iterate on egress listeners, but in cds builder we iterate on sidecarScoped services

**Review by howardjohn — COMMENTED** [review]
are there some test cases we can cover with this

**Comment by ramaraochavali** [comment]
> are there some test cases we can cover with this

+1. I am concerned that our tests did not catch this

**Comment by hzxuzhonghu** [comment]
I will add some cases, wanna hear some feedback beforehand

File: pilot/pkg/model/sidecar_test.go [hunk 1]
```diff
@@ -2496,11 +2497,21 @@ func TestCreateSidecarScope(t *testing.T) {
...
+			sidecarScope := convertToSidecarScope(ps, sidecarConfig, "mynamespace")
+
 			numberListeners := len(sidecarScope.EgressListeners)
 			if numberListeners != configuredListeneres {
 				t.Errorf("Expected %d listeners, Got: %d", configuredListeneres, numberListeners)
 			}
 
+			if sidecarConfig == nil {
+				services := sidecarScope.EgressListeners[0].services
+				if !reflect.DeepEqual(services, sidecarScope.services) {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/sidecar_test.go:2509]: @ramaraochavali @howardjohn take a look at this assert

**Comment by hzxuzhonghu** [comment]
@howardjohn @ramaraochavali any chance to have a look

**Review by ramaraochavali — APPROVED** [review]

**Pull request merged**